### PR TITLE
fix: 鳥取県データのソースに style.json から minzoom/maxzoom/bounds を適用 (#91)

### LINKED ISSUE #91: bug: 鳥取県データのラスタータイルが InvalidStateError で表示されない
## 概要

#89 で追加した鳥取県スマートシティデータのラスタータイルレイヤーを表示すると `InvalidStateError: The source image could not be decoded` エラーが発生し、地図上にデータが表示されない。

## 現在の動作

1. デモ画面で鳥取県データのチェックボックスをONにする
2. MapLibre がタイルをリクエストするが、タイルが存在しない範囲（例: 沖縄付近）では S3 が 404 XML レスポンスを返す
3. MapLibre がその XML を画像としてデコードしようとして `InvalidStateError` が発生する

## 期待する動作

- タイルが存在する範囲・ズームレベルでのみリクエストが発行される
- エラーなくラスタータイルが表示される

## 原因

`addTottoriDataSource` でラスタータイルソースを追加する際に `minzoom` / `maxzoom` / `bounds` を設定していない。各エントリの `styleUrl` が指す style.json にはこれらの情報が含まれている（例: `minzoom: 9`, `maxzoom: 19`）が、現在の実装では利用していない。

## 修正方針

各エントリの `styleUrl` から style.json を取得し、ソース設定（`minzoom`, `maxzoom`, `bounds` 等）を `addTottoriDataSource` に適用する。

## 原因 PR

#89

### PR DESCRIPTION
**Pull request by sugama-satsuki**
Closes #91

## 再現条件

デモ画面（沖縄表示中）で鳥取県スマートシティのデータレイヤーをONにすると、`InvalidStateError: The source image could not be decoded` エラーが発生しデータが表示されない。

## Root Cause

`addTottoriDataSource` でラスタータイルソースを追加する際に `minzoom` / `maxzoom` / `bounds` を設定していなかった。タイルが存在しない範囲へのリクエストで S3 が 404 XML レスポンスを返し、MapLibre がそれを画像としてデコードしようとして失敗していた。

## 修正内容

- `fetchTottoriStyleSourceConfig` 関数を新規追加: 各エントリの `styleUrl` から style.json を取得し、ソース設定（`minzoom`, `maxzoom`, `bounds`）を抽出
- `addTottoriDataSource` を非同期化し、取得した設定をソースに適用
- style.json 取得失敗時はフォールバック（設定なしで追加）

## 影響範囲

鳥取県データの表示機能のみ。既存のハザードマップ・国土数値情報等には影響なし。

## テスト

- `fetchTottoriStyleSourceConfig`: style.json 取得成功 / sources なし / fetch 失敗 の3ケース
- `addTottoriDataSource`: minzoom/maxzoom 適用 / style.json 取得失敗時のフォールバック / 既存ソースの重複チェック
- 全185テストパス

## 原因 PR

#89

## Summary by CodeRabbit

* **新機能**
  * スタイル定義からタイルのズーム範囲と境界を自動取得し、地図ソースに適用するようになりました。
  * ソース構成の取得処理が非同期化され、ソース追加は取得完了を待ってから反映されるようになりました。

* **テスト**
  * 新しい取得ロジックと非同期フローをカバーする単体テストを追加・更新しました。

### CODE REVIEW

File: docs/index.js [hunk 1]
```diff
@@ -4292,24 +4292,68 @@
...
+                const config = {};
+                if (typeof src.minzoom === 'number') {
+                    config.minzoom = src.minzoom;
+                }
+                if (typeof src.maxzoom === 'number') {
+                    config.maxzoom = src.maxzoom;
+                }
+                if (Array.isArray(src.bounds)) {
+                    config.bounds = src.bounds;
+                }
```
[the remark this answers is not in the document]
Comment (sugama-satsuki) [reply docs/index.js]: @coderabbitai 修正しました。`fetchTottoriStyleSourceConfig` に `tileUrl` 引数を追加し、`Object.values(style.sources).find()` で `tileUrl` に一致する source を選択するように変更しました。一致する source がない場合は空オブジェクトを返します。テストも複数 sources のケースを追加しています。

File: docs/index.js [hunk 2]
```diff
@@ -4292,24 +4292,68 @@
...
+                    source.minzoom = styleConfig.minzoom;
+                }
+                if (styleConfig.maxzoom !== undefined) {
+                    source.maxzoom = styleConfig.maxzoom;
+                }
+                if (styleConfig.bounds !== undefined) {
+                    source.bounds = styleConfig.bounds;
+                }
+                map.addSource(id, source);
+                return id;
```
[the remark this answers is not in the document]
Comment (sugama-satsuki) [reply docs/index.js]: @coderabbitai 修正しました。`addTottoriDataSource` を early return パターンに変更し、await 後に `map.getSource(id)` を再チェックしてから `addSource` を呼ぶようにしました。既存ソースがある場合は早期に `id` を返します。

File: src/utils/tottoriDataUtils.ts [hunk 3]
```diff
@@ -46,13 +46,34 @@ export function getTileType(tileUrl: string): 'raster' | 'vector' {
...
+    const style = await fetchJson(styleUrl);
+    if (!style || !style.sources) { return {}; }
+    const sourceKey = Object.keys(style.sources)[0];
+    if (!sourceKey) { return {}; }
+    const src = style.sources[sourceKey];
+    const config: { minzoom?: number; maxzoom?: number; bounds?: number[] } = {};
+    if (typeof src.minzoom === 'number') { config.minzoom = src.minzoom; }
+    if (typeof src.maxzoom === 'number') { config.maxzoom = src.maxzoom; }
+    if (Array.isArray(src.bounds)) { config.bounds = src.bounds; }
+    return config;
```
[the remark this answers is not in the document]
Comment (sugama-satsuki) [reply src/utils/tottoriDataUtils.ts]: @coderabbitai 修正しました。bounds の検証を強化し、`Array.isArray(src.bounds) && src.bounds.length === 4 && src.bounds.every((v) => typeof v === 'number' && Number.isFinite(v))` でチェックするようにしました。型も `[number, number, number, number]` タプルに変更しています。不正な bounds のテストケースも追加しました。

**Comment by sugama-satsuki** [comment]
## 🧂 味付け完了レポート — PR #93

### 📋 概要
コンフリクト解決（main merge）を含め、以下の観点で仕上げを完了しました。

### ✅ 完了項目

**1. コンフリクト解決**
- ✅ ビルド成果物（docs/index.js.map）の再生成により、main マージ時のコンフリクトを解決
- ✅ 本体ロジックの変更は完全に保持（tileUrl 対応 source 選択、TOCTOU race 対策、bounds 検証等）
- ✅ テスト・ビルド・lint 再実行で壊れていないことを確認

**2. CodeRabbit レビューコメント対応**
- ✅ TOCTOU race condition: 既に `await` 後に `map.getSource(id)` を再チェック（L99-101）
- ✅ bounds の型定義: `[number, number, number, number]` の 4-element tuple に固定（L52, 62）
- ✅ bounds の検証: 配列長 4、全て `Number.isFinite` の厳密なチェック（L65-70）
- ✅ tileUrl 対応 source 選択: `Object.entries().find()` で正確にマッチする source を特定（L57-60）
- ✅ テストケース充実: edge case（sources 空、bounds 不正等）も網羅（L150-181）

**3. テスト・ビルド検証**
- ✅ `npm test`: 195 テスト全通過（tottoriDataUtils 含む）
- ✅ `npm run build`: 警告あるが成功

**4. 安全性確認**
- ✅ マップの race condition に対する防御機構が完備
- ✅ バウンダリ値の入力検証が厳密

### 🟢 マージ判定: **提出 OK**
- mergeable: MERGEABLE ✅
- CI/CD: Workers Build SUCCESS ✅
- テスト: 195/195 PASS ✅
- セキュリティ/型安全性: 問題なし ✅

### 📝 作業内容
- コンフリクト解決コミット 1件
- ビルド成果物更新コミット 1件

---
**レビュー観点** — 以下の点で動作を確認してください：
- [ ] 鳥取県データレイヤー ON/OFF で InvalidStateError が出ないこと
- [ ] minzoom/maxzoom/bounds が style.json から正確に反映されていること
- [ ] 複数タイルタイプ（raster/vector）で適切なソースが選択されていること
